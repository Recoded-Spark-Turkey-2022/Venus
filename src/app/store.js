// eslint-disable-next-line no-unused-vars
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import listingReducer from '../features/listings/listingSlice';

export const store = configureStore({
  reducer: {
    listing: listingReducer,
  },
  // eslint-disable-next-line no-shadow
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
