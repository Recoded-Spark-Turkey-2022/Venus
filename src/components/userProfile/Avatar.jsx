import editicon from '../../assets/editicon.svg'

const Avatar = ({name, isOpen, file})=> {


    return(
        <div className="flex flex-col relative justify-center mt-10 m-3 w-48 h-48 ">
            <img className="manImg  bg-openRose" src={file ? URL.createObjectURL(file)
                  : "https://icon-library.com/images/no-image-icon/no-image-icon-0.jpg"} alt="Avatar" />
            <button type='button' className='flex items-center absolute bottom-0 right-0' onClick={isOpen}>
                <img className='bg-mediumBlue rounded-full w-12 h-12' src={editicon} alt="edit-icon" />
                </button>
            <p className="text-center py-2">{name}</p>
        </div>
    )
}

export default Avatar;