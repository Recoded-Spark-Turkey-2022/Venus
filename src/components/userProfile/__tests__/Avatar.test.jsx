import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { store } from '../../../app/store';
import Avatar from '../Avatar';


describe('Avatar', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(
          <Router>
            <Provider store={store}>
              <Avatar />
            </Provider>
          </Router>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });