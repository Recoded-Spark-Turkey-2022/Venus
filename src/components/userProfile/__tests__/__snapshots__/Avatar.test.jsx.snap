// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<div
  className="flex flex-col relative justify-center mt-10 m-3 w-48 h-48 "
>
  <img
    alt="Avatar"
    className="manImg  bg-openRose"
    src="https://icon-library.com/images/no-image-icon/no-image-icon-0.jpg"
  />
  <button
    className="flex items-center absolute bottom-0 right-0"
    type="button"
  >
    <img
      alt="edit-icon"
      className="bg-mediumBlue rounded-full w-12 h-12"
      src="editicon.svg"
    />
  </button>
  <p
    className="text-center py-2"
  />
</div>
`;
